daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/preprocessing/CommandPreprocessor.scala

```diff
@@ -86,13 +86,13 @@ private[preprocessing] final class CommandPreprocessor(compiledPackages: Mutable
       contractKey: Value[Value.AbsoluteContractId],
       choiceId: Ref.ChoiceName,
       argument: Value[Value.AbsoluteContractId],
-  ): speedy.Command.ExerciseByKey = {
+  ): (speedy.Command.ExerciseByKey, Set[Value.AbsoluteContractId]) = {
     val template = unsafeGetTemplate(templateId)
     val choiceArgType = unsafeGetChoiceArgType(templateId, template, choiceId)
     val ckTtype = unsafeGetContractKeyType(templateId, template)
-    val arg = valueTranslator.unsafeTranslateValue(choiceArgType, argument)
-    val key = valueTranslator.unsafeTranslateValue(ckTtype, contractKey)
-    speedy.Command.ExerciseByKey(templateId, key, choiceId, arg)
+    val (arg, argCids) = valueTranslator.unsafeTranslateValue(choiceArgType, argument)
+    val (key, _) = valueTranslator.unsafeTranslateValue(ckTtype, contractKey)
```

Comment:
Minor: Maybe add a comment here that it is safe to discard the result of unsafeTranslateValue because contract keys cannot contain contract IDs and this therefore always must be empty.

Reply:
add an explicit check.